towerlib/towerlib.py

```diff
@@ -1997,24 +2011,579 @@ def schedules(self):
+
+        Args:
+            given_project_id: the id of the project.
+
+        Returns:
+            dict: the project with all the information.
+
+        """
+
+        project_update_list = self.get_project_updates()
+        project_updates = [item for item in project_update_list if item['project'] == given_project_id]
+        return project_updates
+
+    def get_project_update_by_id(self, project_update_id):
+        """Get project update with the given project_update id.
+
+        Args:
+            project_update_id: the id of the project_update.
+
+        Returns:
+            dict: the project_update with all the information.
+
+        """
+
+        project_updates_url = '{api}/project_updates/{id}'.format(api=self.api, id=project_update_id)
+        response = self.session.get(project_updates_url)
+        if not response.ok:
+            self._logger.error("Error getting the project updates. response was: {})".format(response.text))
+            return None
+        else:
+            return response.json()
+
+    def get_all_hosts_from_non_smart_inventories(self):
+        """Get all the hosts from provided that the inventory of the host is not a smart inventory.
+
+        Returns:
+            list: filtered host list.
+
+        """
+        inventory_hosts = [item for item in self.hosts if item.inventory.kind != 'smart']
+        return inventory_hosts
+
+    def get_job_events_by_host(self, host):
+        """Get all the job_events for host.
+
+        Args:
+            host: host EntityManager object.
+
+        Returns:
+            list: list of all the job events for the given host.
+
+        """
+        response = self.session.get('{}/hosts/{}/job_events/'.format(self.api, host.id))
+        if not response.ok:
+            self._logger.error("Error getting the job events. response was: {})".format(response.text))
+            return None
+        else:
+            return response.json().get('results', [])
+
+    def get_job_dates_by_host(self, host):
+        """Get job dates from ansible tower for a given host id when the job event was created.
+
+        Args:
+            host: the host object of tower instance.
+
+        Returns:
+            list: a list containing the formatted datetime objects.
+
+        """
+        self._logger.debug("Start getting job events for the host '{}'".format(host.name))
+        job_events = self.get_job_events_by_host(host)
+        self._logger.debug("Finished getting job events for the host '{}'".format(host.name))
+        if job_events is None:
+            self._logger.error("No job events found.")
+            return None
+        else:
+            self._logger.debug("job events found")
+            job_dates = []
+            for job_event in job_events:
+                job_event_time = job_event['created']
+                job_dates.append(job_event_time)
+            return job_dates
+
+    def get_groups_by_host(self, host):
```

Comment:
The host object already exposes a groups attribute which is an entity manager itself with access to all the underlying information. This method is not needed.